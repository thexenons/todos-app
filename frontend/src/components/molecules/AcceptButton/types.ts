import { ButtonProps } from "../../atoms/Button/types";

export interface AcceptButtonProps {
	disabled?: ButtonProps["disabled"];
	onClick?: ButtonProps["onClick"];
}
