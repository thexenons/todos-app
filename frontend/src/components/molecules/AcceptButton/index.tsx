import { FC } from "react";

import Button from "../../atoms/Button";
import { AcceptButtonProps } from "./types";

const AcceptButton: FC<AcceptButtonProps> = ({ disabled = false, onClick }) => (
	<Button
		variant={Button.variants.contained}
		colorVariant={Button.colorVariants.primary}
		disabled={disabled}
		onClick={onClick}
	>
		Accept
	</Button>
);

export default AcceptButton;
