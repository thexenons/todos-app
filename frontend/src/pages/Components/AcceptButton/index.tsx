import { FC } from "react";

import AcceptButton from "../../../components/molecules/AcceptButton";
import * as SS from "../styled";

const AcceptButtonComponenstPage: FC = () => (
	<SS.ComponentsListWrapper>
		<h1>AcceptButton</h1>
		<SS.ComponentsRowWrapper>
			<AcceptButton />
			<AcceptButton disabled />
		</SS.ComponentsRowWrapper>
	</SS.ComponentsListWrapper>
);

export default AcceptButtonComponenstPage;
