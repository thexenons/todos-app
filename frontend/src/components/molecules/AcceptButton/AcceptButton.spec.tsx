import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import AcceptButton from ".";

describe("AcceptButton", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<AcceptButton />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
