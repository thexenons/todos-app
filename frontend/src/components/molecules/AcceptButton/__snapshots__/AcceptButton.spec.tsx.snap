// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AcceptButton > Render 1`] = `
<div>
  <button
    class="styled__ButtonWrapper-sc-f8cd241f-0 bYSNlB"
  >
    Accept
  </button>
</div>
`;
